src/main/java/core/basesyntax/service/strategy/strategyimpl/BalanceOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        Storage.storage.put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
This is a direct manipulation of the storage from within the operation class. While this is not incorrect, consider encapsulating the access to the storage within a service layer to follow the SOLID principles better.